src/run.ts

```diff
@@ -61,18 +74,33 @@ export const run = async (context: Context) => {
     }
 
     setFailed(message);
+    return;
   }
 
   let flags = getInput("jiraIDFlags");
   flags = flags.includes("g") ? flags : flags + "g";
   const regex2 = RegExp(getInput("jiraIDRegexp"), flags);
 
+  const skipIDs = getInput("jiraSkipCheck")?.split(",") || [];
+  const ids = [...pullRequestTitle.matchAll(regex2)][0];
+  if (ids.find((id: string) => skipIDs.includes(id.toLowerCase()))) {
+    return;
+  }
+
   if (
     !pullRequestTitle ||
-    !(await containsParentTaskID([...pullRequestTitle.matchAll(regex2)]))
+    ids.length === 0 ||
+    !(await containsParentIssueID(ids))
   ) {
-    const message = `Pull Request title "${pullRequestTitle}" doesn't contain JIRA parent task ID
+    const message = `Pull Request title "${pullRequestTitle}" doesn't contain JIRA parent issue ID
 `;
     setFailed(message);
+    return;
+  }
+
+  if (await isIssueSpike(ids)) {
```

Comment:
this algorithm doesn't seem to be optimal as it is calling JIRA twice (once for checking for parent, once for checking for issue type). Why not to store the data from the first call and just do a check in-memory on the issue type as you are already obtained the data in the first call?

Reply:
that is correct, i wanted to do it that way initially but doing so would require larger refactor and given the purpose of this action and time spent on trying to implement it that way did not seem appropriate